lib/surface/base_component.ex

```diff
@@ -19,8 +19,8 @@ defmodule Surface.BaseComponent do
 
   @optional_callbacks transform: 1
 
-  defmacro __using__(opts \\ []) do
-    type = Keyword.get(opts, :type)
+  defmacro __using__(opts) do
+    type = Keyword.fetch!(opts, :type)
```

Comment:
As the `:type` option is required, moving from `get` to `fetch!` will give a better message in case the developer does not provide the option.